### PR TITLE
`deer`: update metadata

### PR DESCRIPTION
**Pull request by indietyp**
## 🌟 What is the purpose of this PR?

Update the `README.md` and `Cargo.toml` to prepare for the initial reservation of crate names.

### CODE REVIEW

**Review by thehabbos007 — APPROVED** [review]
Awesome, thanks! Looks good from my side. David should approve legal stuff before we merge :)

File: libs/deer/desert/README.md
Comment (TimDiekmann) [inline libs/deer/desert/README.md:3]:
```suggestion
`desert` is the internal only deserialization testing framework used throughout the integration tests and should never
```

File: libs/deer/README.md
Comment (TimDiekmann) [inline libs/deer/README.md:10]: I can add this section when publishing the crates, we don't need it in the repository I guess
```suggestion
```

File: libs/deer/json/README.md
Comment (TimDiekmann) [inline libs/deer/json/README.md:12]:
```suggestion
```

File: libs/deer/macros/README.md
Comment (TimDiekmann) [inline libs/deer/macros/README.md:12]:
```suggestion
```

**Review by TimDiekmann — APPROVED** [review]

**Review by TimDiekmann — APPROVED** [review]

**Review by nonparibus — APPROVED** [review]

**Pull request merged**
